src/index/updater/brc721_updater.rs

```diff
@@ -14,189 +14,207 @@
+			return Ok(());
 		}
-	}
-
-	#[test]
-	fn test_one_collection() {
-		let expected_height = 100u32;
-		let expected_rebaseable = true;
-		let expected_tx_index = 5;
-
-		let mut id_to_collection = HashMap::new();
-
-		let mut updater =
-			Brc721Updater { height: expected_height, collection_table: &mut id_to_collection };
-
-		let tx = brc721_collection_tx(expected_rebaseable);
-
-		updater.index_collections(expected_tx_index, &tx).unwrap();
-
-		assert_eq!(id_to_collection.len(), 1);
-		let key = (expected_height.into(), expected_tx_index);
-		assert!(id_to_collection.contains_key(&key));
 
-		let (address, rebaseable) = id_to_collection.get(&key).unwrap();
-		assert_eq!(*address, COLLECTION_ADDRESS);
-		assert_eq!(*rebaseable, expected_rebaseable);
-	}
-
-	#[test]
-	fn test_no_collections() {
-		let expected_height = 100u32;
-		let mut id_to_collection = HashMap::new();
-
-		let mut updater =
-			Brc721Updater { height: expected_height, collection_table: &mut id_to_collection };
-
-		let tx_index = 5;
-		let tx = empty_tx();
+		let tx_info = if let Ok(tx_info) =
+			self.client.get_raw_transaction_info(&tx.input[0].previous_output.txid, None)
+		{
+			tx_info
+		} else {
+			return Ok(());
+		};
 
-		updater.index_collections(tx_index, &tx).unwrap();
+		let script_buf = tx_info.vout[tx.input[0].previous_output.vout.into_usize()]
+			.script_pub_key
+			.script()?;
 
-		assert_eq!(id_to_collection.len(), 0);
-	}
+		let network = match self.client.get_blockchain_info() {
+			Ok(blockchain_info) => blockchain_info.chain,
+			_ => Network::Bitcoin,
+		};
 
-	#[test]
-	fn test_multiple_transactions() {
-		let expected_height = 100u32;
-		let mut id_to_collection = HashMap::new();
+		let address = if let Ok(address) = Address::from_script(&script_buf, network) {
+			address
+		} else {
+			return Ok(());
+		};
 
-		let mut updater =
-			Brc721Updater { height: expected_height, collection_table: &mut id_to_collection };
+		// Get the h160 address contained in the first input or return Ok if it's not
+		// P2PKH/P2WPKH/P2TR(nothing to index)
+		let h160_address = if let Ok(address) = btc_address_to_h160(address) {
+			address
+		} else {
+			return Ok(());
+		};
+
+		for (index, slot_bundle) in register_ownership.slots_bundles.into_iter().enumerate() {
+			for slot_range in slot_bundle.0.into_iter() {
+				let owner_bytes = tx.output[index + 1].clone().script_pubkey.into_bytes();
+				let hex_encoded_owner = hex::encode(&owner_bytes);
+				let slot_range_owned_by_owner =
+					self.tokens_for_owner.get_value(hex_encoded_owner.clone()).unwrap_or(0);
+
+				let slot_start = *slot_range.start();
+				let slot_end = *slot_range.end();
+
+				for slot in slot_range {
+					let mut slot_bytes = [0u8; 12];
+					slot_bytes.copy_from_slice(&slot.to_le_bytes()[..12]);
+					let token_id = (slot_bytes, h160_address.0);
+					if self.token_owners.get_value((token_id, collection_id_value)).is_some() {
```

Comment:
consider the use of `slot_and_owner_to_token_id` from here: https://github.com/freeverseio/laos-btc/blob/43b2b797916e2e8fd1a0d9c4bb3893b3e2eee613/crates/ordinals/src/brc721/token_id.rs#L58